tracer/src/Datadog.Trace/AppSec/RcmModels/EncoderExtensions.cs

```diff
@@ -0,0 +1,28 @@
+// <copyright file="EncoderExtensions.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Collections.Generic;
+using System.Linq;
+using Datadog.Trace.AppSec.RcmModels.Asm;
+using Datadog.Trace.AppSec.RcmModels.AsmData;
+using Datadog.Trace.AppSec.Waf;
+using Datadog.Trace.AppSec.Waf.NativeBindings;
+
+namespace Datadog.Trace.AppSec;
+
+internal static class EncoderExtensions
+{
+    internal static Obj Encode(this List<RuleData> ruleData, WafLibraryInvoker wafLibraryInvoker)
+    {
+        var dictionary = new Dictionary<string, object> { { "rules_data", ruleData.Select(r => r.ToKeyValuePair()).ToArray() } };
+        return Encoder.Encode(dictionary, wafLibraryInvoker, new List<Obj>(), false);
```

Comment:
I would set the third parameter to `null` rather than `new List<Obj>()`. We sometimes need to dispose the objects generated as result of encoding (only when encoding the waf rules, I think), sometimes we don't. If we don't want to store them passing null will mean they are ignored, passing empty list will mean they're added to the list and later the list thrown away by the GC.